### PR TITLE
Mwpw-147034: Custom border color + badge/border color decoupling [Mer…

### PR DESCRIPTION
**Pull request by Axelcureno**
Adds ability to decouple border and badge colors in authoring of merch card.
Adds ability to set custom border color without using a badge in merch card.

Resolves: [MWPW-147034](https://jira.corp.adobe.com/browse/MWPW-147034)

Milo PR: https://github.com/adobecom/milo/pull/2613

Test URLs:

Before: https://main--milo--adobecom.hlx.page/drafts/axel/custom-border-badge?martech=off
After: https://mwpw-147034--milo--adobecom.hlx.page/drafts/axel/custom-border-badge?martech=off

- https://mwpw-147034--mas--adobecom.hlx.page/

### CODE REVIEW

**Review by npeltier — APPROVED** [review]
lgtm

File: web-components/src/merch-card.js [hunk 1]
```diff
@@ -110,10 +111,12 @@ export class MerchCard extends LitElement {
 
     updated(changedProperties) {
         if (
-            changedProperties.has('badgeBackgroundColor') &&
+            (changedProperties.has('badgeBackgroundColor') || changedProperties.has('borderColor')) &&
```
Comment (yesil) [inline web-components/src/merch-card.js:114]: IMO this desserves a unit test. Logic is getting complicated.

**Review by yesil — CHANGES_REQUESTED** [review]

File: web-components/src/merch-card.js [hunk 2]
```diff
@@ -110,10 +111,12 @@ export class MerchCard extends LitElement {
...
             this.variant !== 'twp'
         ) {
-            this.style.border = `1px solid ${this.badgeBackgroundColor}`;
+            this.style.border = `1px solid ${
+                this.borderColor ? this.borderColor : this.badgeBackgroundColor
+            }`;
```
Comment (yesil) [inline web-components/src/merch-card.js:119]: also would it be possible to move this in a getter?

File: web-components/src/merch-card.js (hunk 1 above)
Comment (Axelcureno) [reply web-components/src/merch-card.js]: unit test added

File: web-components/src/merch-card.js (hunk 2 above)
Comment (Axelcureno) [reply web-components/src/merch-card.js]: Moved to getter `computedBorderStyle()`

**Review by mirafedas — APPROVED** [review]

**Pull request merged**